baseclasses/pyAero_problem.py

```diff
@@ -551,19 +629,71 @@ def addDV(
                 "(...,alpha=value, ...) must be given. The list of "
                 "possible DVs are: %s." % (key, repr(self.possibleDVs))
             )
-        if key in self.possibleBCDVs:
-            if family is None:
-                raise Error("The family must be given for BC design variables")
+
+        if key in self.possibleBCDVs or key in self.possibleActuatorDVs:
+            if familyGroup is None:
+                raise Error(
+                    "The familyGroup must be given for BC or actuator\
+                             design variables"
+                )
 
             if name is None:
-                dvName = "%s_%s_%s" % (key, family, self.name)
+                dvName = "%s_%s_%s" % (key, familyGroup, self.name)
             else:
                 dvName = name
 
             if value is None:
-                if (key, family) not in self.bcVarData:
-                    raise Error("The value must be given or set using the setBCVar routine")
-                value = self.bcVarData[key, family]
+                if key in self.possibleBCDVs:
+                    try:
+                        value = self.bc_data[familyGroup][key]
+                    except KeyError:
+                        raise Error("The value must be given or set using the setBCVar routine")
+                else:
+                    try:
+                        value = self.actuatorData[familyGroup][key]
+                    except KeyError:
+                        raise Error("The value must be given or set using the setActuatorVar routine")
+
+            # # the value of the BCData[familyGroup][key] maybe a dictionary of data for each patch
```

Comment:
Is this still needed? I got the point codewise, but I am wondering if we need  or want to allow this kind of nested dictionaries in the BCdata definition